invenio_banners/services/results.py

```diff
@@ -66,7 +66,7 @@ def __init__(
     @property
     def hits(self):
         """Iterator over the hits."""
-        for record in self._results:
+        for record in self.banners_result():
```

Comment:
I agree this would look better!  But using the parent's `hits` in this situation will not work as banners don't have a UUID field (going deeper into code this problem will appear)

Reply:
No worries, I didn't check too deep either, but it seemed like a decent bet if it meant removing code :)